main.go

```diff
@@ -136,23 +179,48 @@ func main() {
 
 	// Do feature discovery from all configured sources.
 	for _, source := range sources {
-		labelsFromSource, err := getFeatureLabels(source)
-		if err != nil {
-			stderrLogger.Printf("discovery failed for source [%s]: %s", source.Name(), err.Error())
-			stderrLogger.Printf("continuing ...")
-			continue
+
+		var labelsFromSource Labels
+		if _, enabled := enabledResourceTypes[TypeBinary]; enabled {
+			labelsFromSource, err = getFeatureLabels(source)
```

Comment:
May be we can change `getFeatureLabels()` -> `getBinaryResources()`. Elsewhere we can document that binary resources are published as labels and integer resources are advertised by patching the capacity of the cluster. Just a thought. LMK WDYT?